packages/logging/src/analytics-helpers.ts

```diff
@@ -351,31 +349,28 @@ export class ThrottledAnalytics implements MongoshAnalytics {
     return this.throttleState.count < this.throttleOptions.rate;
   }
 
-  flush(callback: (err?: Error | undefined) => void): void {
+  async flush(): Promise<void> {
     if (!this.throttleOptions) {
-      this.target.flush(callback);
+      await this.target.flush();
       return;
     }
 
     if (!this.currentUserId) {
-      callback(
-        new Error('Trying to persist throttle state before userId is set')
-      );
-      return;
+      throw new Error('Trying to persist throttle state before userId is set');
     }
 
-    this.restorePromise.finally(async () => {
-      try {
-        await fs.promises.writeFile(
-          this.metadataPath,
-          JSON.stringify(this.throttleState)
-        );
-        await this.unlock();
-        this.target.flush(callback);
-      } catch (e) {
-        callback(e as Error);
-      }
-    });
+    try {
+      await this.restorePromise;
+    } catch (err) {
```

Comment:
```suggestion
    } catch {
```

Doesn't the linter complain about an unused variable otherwise?